src/3D/Camera.h

```diff
@@ -26,11 +26,22 @@ class Camera
 public:
 	Camera(glm::vec3 position, glm::vec3 rotation)
 	    : _position(position)
+	    , _viewCenterLand(_position)
 	    , _rotation(glm::radians(rotation))
+	    , _dwv(0.0f, 0.0f, 0.0f)
 	    , _dv(0.0f, 0.0f, 0.0f)
+	    , _drv(0.0f, 0.0f, 0.0f)
 	    , _projectionMatrix(1.0f)
+	    , _wVelocity(0.0f, 0.0f, 0.0f)
 	    , _velocity(0.0f, 0.0f, 0.0f)
+	    , _rotVelocity(0.0f, 0.0f, 0.0f)
+	    , _groundHeight(0.0f)
+	    , _groundHeightStart(0.0f)
+	    , _movementSpeed(4.0f)
+	    , _accelFactor(0.001f)
+	    , _rotAccelFactor(0.001f)
 	    , _maxMovementSpeed(0.005f)
+	    , _maxRotationSpeed(0.005f)
 	    , _freeLookSensitivity(1.0f)
 	{
 	}
```

Comment:
I know you said you're a little rusty in C++, but I think this is a good opportunity to split declaration and definition of the camera constructor.

What I mean is that we have this in the header:
```c++
Camera(glm::vec3 position, glm::vec3 rotation);
```
and this in the .cpp implementation:
```c++
Camera::Camera(glm::vec3 position, glm::vec3 rotation)
	    : _position(position)
	    , _viewCenterLand(_position)
	    , _rotation(glm::radians(rotation))
	    , _dwv(0.0f, 0.0f, 0.0f)
	    , _dv(0.0f, 0.0f, 0.0f)
	    , _drv(0.0f, 0.0f, 0.0f)
	    , _projectionMatrix(1.0f)
	    , _wVelocity(0.0f, 0.0f, 0.0f)
	    , _velocity(0.0f, 0.0f, 0.0f)
	    , _rotVelocity(0.0f, 0.0f, 0.0f)
	    , _groundHeight(0.0f)
	    , _groundHeightStart(0.0f)
	    , _movementSpeed(4.0f)
	    , _accelFactor(0.001f)
	    , _rotAccelFactor(0.001f)
	    , _maxMovementSpeed(0.005f)
	    , _maxRotationSpeed(0.005f)
	    , _freeLookSensitivity(1.0f)
	{
	}
```

Reply:
Oh and make sure the order of initialization is the same as declaration (see the warnings in diff)

Reply:
I think I've done this right, you will need to check.